binding/ios/PorcupineManager.swift

```diff
@@ -0,0 +1,131 @@
+//
+// Copyright 2018 Picovoice Inc.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import AudioToolbox
+import AVFoundation
+import Foundation
+
+import pv_porcupine
+
+/// Errors corresponding to different status codes returned by Porcupine library.
+enum PorcupineManagerError: Error {
+    case OutOfMemoryError
+    case IOError
+    case InvalidArgumentError
+}
+
+/// iOS binding for Picovoice's wake word detection engine (aka Porcupine).
+class PorcupineManager {
+    private let numBuffers = 5
+    private let modelFilePath: String
+    private let keywordFilePath: String
+    private let sensitivity: Float
+    private let keywordCallback: (() -> Void)
+
+    private var isListening = false
+    private var queue: AudioQueueRef?
+    private var porcupine: OpaquePointer?
+
+    private let audioCallback: AudioQueueInputCallback = {
+        userData, queue, bufferRef, startTimeRef, numPackets, packetDescriptions in
+
+        let porcupineManager: PorcupineManager = Unmanaged<PorcupineManager>.fromOpaque(userData!).takeUnretainedValue()
+        let pcm = bufferRef.pointee.mAudioData.assumingMemoryBound(to: Int16.self)
+        var result = false
+
+        pv_porcupine_process(porcupineManager.porcupine, pcm, &result)
```

Comment:
Should we check the status returned by pv_porcupine_process and throw accordingly?